### PR TITLE
refactor: replace user store with observables [WEB-799]

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description
All automated tests pass and I haven't noticed any failures with manual testing.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/stores/users.tsx [hunk 1]
```diff
@@ -13,230 +14,108 @@ type UsersPagination = {
...
+      return Loaded(userPage);
+    });
   };
 
-  return Loaded(userPage);
-};
+  public ensureUsersFetched = async (
+    canceler: AbortController,
+    cfg?: FetchUsersConfig,
+  ): Promise<void> => {
```
Comment (ClaireNeveu) [inline webui/react/src/stores/users.tsx:79]: My one note is that in idiomatic reactive programming, _actions_ (fetch function in the case of stores) should always return `void`. Exposing the promise here allows consumers to await the promise when they should instead be subscribing to the observables returned by the various `get` functions.

Comment (EmilyBonar) [reply webui/react/src/stores/users.tsx:79]: That's a good point, I'll go remove those promises.

**Review by gt2345 — APPROVED** [review]
LGTM, one side note is that some parts of the app import from 'micro-observables' and other parts import from 'utils/observable'.

**Comment by EmilyBonar** [comment]
I'm going to wait on merging this until the release is cut so that we're not rushing to fix bugs in the event that we missed something.

**Pull request merged**